### DISCUSSION TITLE
How to handle Authorization header using localStorage?

### DISCUSSION THREAD

**Discussion by kmcalpine** [vikejs/vike#1063] [Discussion]
Currently I am storing the bearer token in a cookie, then in the server I add the token to the pageContext so it can be accessed throughout the various pages. Although I would much rather store the token in localStorage so I don't need to bother with XSRF. I've tried using custom exports but localStorage is not defined (index.page.tsx), or the export is not visible to the server page (index.page.client.tsx). Any help is appreciated, thanks!

**Comment by brillout** [comment]
Seems like you found a solution. Let me know if you still have questions.

(Also, in case your company is up for it: https://github.com/sponsors/brillout.)

**Comment by kmcalpine** [comment]
@brillout Hi, I thought I found a solution, but unfortunately its not working. Basically, I'm having difficulty in storing a token in localStorage, then making it accessible to my server pages for server sided data fetching of authenticated endpoints, any help would be great, thanks!

**Reply by brillout** [reply]
I'm not sure what you want is even possible: the `localStorage` data cannot be accessed for the first HTML render.

If I were you I'd investigate using a cookie instead.
